### ISSUE TITLE
[Data Builders] Fields with @skip or @include Directives should be nullable

### ISSUE THREAD

**Issue by matthewrkula** [apollographql/apollo-kotlin#4639]
Apollo generated kotlin classes properly reflect that a field is nullable when it has a `@skip` or `@include` directive but Data Builders currently do not.

For example if I have a sample query like this

```graphql
query Sample($useFirst: Boolean!) {
  first @include(if: $useFirst)
  second @skip(if: $useFirst)
}
```
Then the associated Data Builder's `first` and `second` field will not be nullable making it impossible to test scenario's where one or both are null.

**Comment by martinbonnin** [comment]
Good find! Indeed currently if a value is omitted, the default resolver will put a value there...

If Kotlin had union types, we could do something like this:

```kotlin
public class QueryBuilder : ObjectBuilder {
  public var first: Int | Absent
  public var second: Int | Absent
}
```

which could then be used like so:

```kotlin
SampleQuery.Data {
  first = Absent
  second = 42
}
```

But I don't think this is really feasible so making everything nullable might be the solution. We'll dig into this.

**Comment by matthewrkula** [comment]
Yeah that should work! Thanks for the quick response!

**Comment by BoD** [comment]
This has been released as part of [v3.7.5](https://github.com/apollographql/apollo-kotlin/releases/tag/v3.7.5)
